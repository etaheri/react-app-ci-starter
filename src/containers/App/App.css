.App {
  text-align: center;
}

.App-header {
  background-color: #2d2f4e;
  padding: 10rem;
  color: white;
}

.hello {
  animation: hello-spin 1s infinite;
  position: absolute;
  margin: 0 1rem;
}

@keyframes hello-spin {
  0% {
    transform: rotate(0deg);
  }

  50% {
    transform: rotate(20deg);
  }

  100% {
    transform: rotate(0deg);
  }
}
