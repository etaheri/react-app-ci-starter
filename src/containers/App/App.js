import React from 'react';
import './App.css';

const App = () => (
  <div className="App">
    <div className="App-header">
      <h2>
        Welcome to React CI Starter{' '}
        <span role="img" aria-label="wave" className="hello">
          ðŸ‘‹
        </span>
      </h2>
    </div>
  </div>
);

export default App;
