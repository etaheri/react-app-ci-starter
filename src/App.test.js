import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import App from './App';

describe('App Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div'); // eslint-disable-line no-undef
    ReactDOM.render(<App />, div);
  });

  it('it renders a title', () => {
    const renderedComponent = shallow(<App />);
    expect(renderedComponent.find('h2').length).toBe(1);
  });

  it('renders a snapshot', () => {
    const tree = ReactTestRenderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
