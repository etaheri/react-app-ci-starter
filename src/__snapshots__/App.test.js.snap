// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders a snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <h2>
      Welcome to React CI Starter
       
      <span
        aria-label="wave"
        className="hello"
        role="img"
      >
        ðŸ‘‹
      </span>
    </h2>
  </div>
</div>
`;
